polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -2139,47 +2139,52 @@ async fn provide_candidate_to_grid<Context>(
 
 // Utility function to populate per relay parent `ParaId` to `GroupIndex` mappings.
 fn determine_groups_per_para(
-	availability_cores: &[CoreState],
-	group_rotation_info: &GroupRotationInfo,
-	maybe_claim_queue: &Option<ClaimQueueSnapshot>,
+	availability_cores: Vec<CoreState>,
+	group_rotation_info: GroupRotationInfo,
+	maybe_claim_queue: Option<ClaimQueueSnapshot>,
```

Comment:
Why are references no longer working?

Reply:
See above, I think it would be best to call the runtime APIs from this function directly.

Reply:
> Why are references no longer working?

they work smoothly but no longer needed. I'll move the runtime api inside this fn.